### PR TITLE
Add Feature - Page Preview for Simulator

### PR DESCRIPTION
**Pull request by jooshkins**
This PR is related to this feature request:
[[Feature Request] Ability to change simulator background color #1790](https://github.com/inkstitch/inkstitch/issues/1790)

Basically the idea is for the simulator to replicate Inkscape's page view functionality. I was able to get the page color, page size, and desk color to work. 

Initially I tried to replicate the border and drop shadow as well, but ran into issues with getting the drop shadow to show up in the simulator view. So I left out both of those settings.

### CODE REVIEW

**Comment by jooshkins** [comment]
Got the dropshadow and border to show up correctly.

**Comment by lexelby** [comment]
Cool!!

File: lib/api/stitch_plan.py [hunk 1]
```diff
@@ -17,9 +17,17 @@ def get_stitch_plan():
...
+        "height": metadata.document.get('height'),
+        "pagecolor": metadata.document[1].get('pagecolor'),
+        "deskcolor": metadata.document[1].get('inkscape:deskcolor')
+    }
+
     collapse_len = metadata['collapse_len_mm']
     min_stitch_len = metadata['min_stitch_len_mm']
     patches = g.extension.elements_to_stitch_groups(g.extension.elements)
-    stitch_plan = stitch_groups_to_stitch_plan(patches, collapse_len=collapse_len, min_stitch_len=min_stitch_len)
+    stitch_plan = stitch_groups_to_stitch_plan(patches, collapse_len=collapse_len, min_stitch_len=min_stitch_len, page_specs=page_specs)
```
Comment (lexelby) [inline lib/api/stitch_plan.py:31]: I'd rather if we didn't wedge the page specs into the stitch plan.  It doesn't feel like they belong there.

How about a separate API call to fetch the page specs?

Comment (jooshkins) [reply lib/api/stitch_plan.py:31]: That's a good idea, I'll figure it out.

Comment (jooshkins) [reply lib/api/stitch_plan.py]: Moved to separate api call

File: lib/stitch_plan/stitch_plan.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def __json__(self):
                     num_trims=self.num_trims,
                     num_stitches=self.num_stitches,
                     bounding_box=self.bounding_box,
-                    estimated_thread=self.estimated_thread
+                    estimated_thread=self.estimated_thread,
```
Comment (lexelby) [inline lib/stitch_plan/stitch_plan.py:149]: This change can probably go away now too.

File: lib/api/stitch_plan.py [hunk 3]
```diff
@@ -17,6 +17,7 @@ def get_stitch_plan():
         return dict(colors=[], stitch_blocks=[], commands=[])
 
     metadata = g.extension.get_inkstitch_metadata()
+
```
Comment (lexelby) [inline lib/api/stitch_plan.py:20]: Let's roll back this change, no need for a blank line here now.

**Review by lexelby — APPROVED** [review]

**Comment by lexelby** [comment]
Looks good!  I'll merge this once you're ready, just let me know.

Thank you very much for your contribution!  If you're interested in doing more in the future, I can send you an invite to be a collaborator on this repo, which would let you push branches here and use our CI.  If that sounds interesting to you, say the word and I'll make it happen.

File: lib/stitch_plan/stitch_plan.py (hunk 2 above)
Comment (jooshkins) [reply lib/stitch_plan/stitch_plan.py]: Removed in latest commit.

File: lib/api/stitch_plan.py (hunk 3 above)
Comment (jooshkins) [reply lib/api/stitch_plan.py]: Removed in latest commit.

**Comment by jooshkins** [comment]
Ready!
I would be interested in collaborating in the future, mostly I think I would be focusing on Electron / front-end development since that is what I have most experience with.
Thanks for the review and all the work you have put into this project!

**Comment by jooshkins** [comment]
Added a bug fix for where the page would be cut off before reaching the toolbar
image

**Comment by kaalleen** [comment]
Thank you for your work on this. This is a very useful feature.
I merged your branch into main. Also I send you an invitation to the Ink/Stitch repository.

**Comment by lexelby** [comment]
Oops, thanks kaalleen, life intervened :sweat_smile:

**Pull request merged**